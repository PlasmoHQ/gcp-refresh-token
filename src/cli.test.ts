import { expect, test } from "@jest/globals"
import { execFileSync } from "child_process"
import { existsSync } from "fs"
import { join } from "path"
import { cwd, execPath } from "process"

const cliScript = join(cwd(), "dist", "cli.js")

const cliErrorRun = () => {
  try {
    execFileSync(execPath, [cliScript]).toString("utf-8")
  } catch (error) {
    console.log(error.message)

    return error.message.split("\n")[0]
  }
}

test("key checking corrects", async () => {
  expect(cliErrorRun()).toMatchSnapshot()

  expect(existsSync(join(cwd(), "key.json"))).toBe(true)

  expect(cliErrorRun()).toMatchSnapshot()
})
