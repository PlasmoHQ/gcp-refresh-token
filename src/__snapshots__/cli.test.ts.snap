// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`key checking corrects 1`] = `"ðŸ›‘ Error: No key.json found, a default key was created. Follow the readme for more info: https://github.com/plasmo-corp/gcp-refresh-token"`;

exports[`key checking corrects 2`] = `"ðŸ›‘ Error: key.json is missing one or more of the required fields: client_id, client_secret, redirect_uris"`;
